File: Review-00-Foundational-Concepts/Templates-Generics-Python/template_demo.py
from __future__ import annotations
from typing import Any

def print_multiple_times(value: Any, count: int):
    """
    Print a value multiple times.

    @param <T> type to print multiple times

    @param value value to print
    @param count number of times to print
    """

    for _ in range(0, count):
        print(f"{value} ", end="")

    print()


def main():
    print_multiple_times(7, 3)
    print()
    print_multiple_times(5.5, 2)
    print()
    print_multiple_times("Hello", 4)
    print()
    print_multiple_times('?', 20)

if __name__ == "__main__":
    main()

File: Review-00-Foundational-Concepts/Templates-Generics-Java/TemplateDemo.java

public class TemplateDemo
{

	/**
	 * Print a value multiple times.
	 *
	 * @param <T> type to print multiple times
	 *
	 * @param value value to print
	 * @param count number of times to print
	 */
	public static <T> void printMultipleTimes(T value, final int count)
	{
		for (int i = 0; i < count - 1; i++) {
			System.out.printf("%s ", value);
		}
        System.out.printf("%s%n", value);
	}


	public static void main(String... argv)
	{
		printMultipleTimes(7, 3);
		System.out.println();
		printMultipleTimes(5.5, 2);
		System.out.println();
		printMultipleTimes(5.5f, 2);
		System.out.println();
		printMultipleTimes("Hello", 4);
		System.out.println();
		printMultipleTimes('?', 20);
		System.out.println();
	}
}
